rockcraft/environment.py

```diff
@@ -0,0 +1,57 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utilities to expand variables in project environments."""
+
+from __future__ import annotations
+
+import pathlib
+from typing import Any
+
+import craft_parts
+
+
+def expand_environment(
```

Comment:
reviewer note, this was extracted from https://github.com/canonical/rockcraft/pull/366/commits/6d29a2b6f1bc83ac0fd7c3719cd417218adf9943#diff-2e78c02a1430e75e37cceef3157f4c33ebc785b8693642ebeda5b5d118ec5935L326